### PR TITLE
feat: retry count in url

### PR DESCRIPTION
**Pull request by pauldambra**
During an incident we are sometimes flooded with retries from the JS SDK.

Or at least we believe we are. And, when we are we don't know how many retries there are.

Let's add the retry count to the URL so we can figure things out by inspecting the capture logs

Tested locally that I can still ingest events etc

### CODE REVIEW

File: src/send-request.ts [hunk 1]
```diff
@@ -17,7 +20,7 @@ export const addParamsToURL = (
         const params = halves[1].split('&')
         for (const p of params) {
             const key = p.split('=')[0]
-            if (args[key]) {
+            if (key in args) {
```
Comment (pauldambra) [inline src/send-request.ts:23]: we can send `ip=0` into this method which would never pass this test

File: src/send-request.ts [hunk 2]
```diff
@@ -29,7 +32,7 @@ export const addParamsToURL = (
 
 export const encodePostData = (data: PostData | Uint8Array, options: Partial<XHROptions>): string | BlobPart | null => {
     if (options.blob && data.buffer) {
-        return new Blob([data.buffer], { type: 'text/plain' })
+        return new Blob([_isUint8Array(data) ? data : data.buffer], { type: 'text/plain' })
```
Comment (pauldambra) [inline src/send-request.ts:35]: our JS tests were using snapshots here. And the snapshot was happily passing when this method returned the empty array for Uint8Arrays passed in

**Comment by pauldambra** [comment]
@daibhin @benjackwhite it's best to go through this one a commit at a time with hide whitespace on 🙈

**Comment by pauldambra** [comment]
Screenshot 2023-10-27 at 13 14 31

the moving around of things is partly because we have a big chunk of the bundle in `utils.ts` and can't see what is making up the size. As I split it up we'll be able to more easily see what we could optimise

**Review by daibhin — APPROVED** [review]

**Pull request merged**